File: AssetManagement.Application/Frontend/src/app/models/asset/AssetCreationRequest.ts
import { FieldValues } from "react-hook-form";
import { AssetStateEnum } from "../../types/enum";

export interface AssetCreationRequest {
    name: string,
    categoryId: string,
    specification: string,
    installedDate: Date,
    state: AssetStateEnum
}

export interface AssetCreationForm extends AssetCreationRequest, FieldValues {}

File: AssetManagement.Contracts/Dtos/AssetDtos/Requests/AssetCreationRequest.cs
using AssetManagement.Domain.Enums;
using System.ComponentModel.DataAnnotations;

namespace AssetManagement.Contracts.Dtos.AssetDtos.Requests;
public class AssetCreationRequest
{
    [Required]
    public string Name { get; set; } = string.Empty;
    [Required]
    public Guid CategoryId { get; set; }
    [Required]
    public string Specification { get; set; } = string.Empty;
    [Required]
    public DateTime InstalledDate { get; set; }
    [Required]
    [EnumDataType(typeof(AssetState))]
    public AssetState State { get; set; }
}
